File: PGProblemSolving/Mathematics/점프와순간이동_12980_Lv2.cpp
//
//  점프와순간이동_12980_Lv2.cpp
//  ProgrammersAlgorithmPractice
//
//  Created by MinKyeongTae on 08/10/2019.
//  Copyright © 2019 Min Kyeong Tae. All rights reserved.
//

/// MARK: - 점프와 순간이동_12980_Lv2

#include <iostream>
using namespace std;

int jumpAndTeleport(int n) {
    int Ans=0;
    while(n>0) {
        if(n%2==0) n/=2;
        else {
            n--;
            Ans++;
        }
    }
    return Ans;
}

File: PGProblemSolving/Mathematics/점프와순간이동_12980_Lv2.swift
//
//  점프와순간이동_12980_Lv2.swift
//  ProgrammersAlgorithmPractice
//
//  Created by MinKyeongTae on 08/10/2019.
//  Copyright © 2019 Min Kyeong Tae. All rights reserved.
//

import Foundation

func jumpAndTeleport(_ n: Int) -> Int {
    var Ans = 0
    var N = n
    while N != 0 {
        if N % 2 == 0 {
            N /= 2
        } else {
            N -= 1
            Ans += 1
        }
    }
    return Ans
}
